### PR TITLE
[WIP]  Ceph: Replicated Pools size are ignored.

### LINKED ISSUE #3144: Edits to CephFilesystem pools replicated size are ignored
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
Changing the metadataPool and dataPools replicated size from 1 to 2 has no effect.

**Expected behavior:**
After editing the CephFilesystem config the dashboard should show the new replica size for the pools, like it does when editing the replicated size of a CephBlockPool.

**How to reproduce it (minimal and precise):**

**Environment**:
* OS (e.g. from /etc/os-release): 18.04.2 LTS
* Kernel (e.g. `uname -a`): Linux areed-aka-8tw9 4.15.0-1030-gcp #32-Ubuntu SMP Wed Apr 10 10:27:45 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
* Cloud provider or hardware configuration: GCP
* Rook version (use `rook version` inside of a Rook Pod): v1.0.0
* Kubernetes version (use `kubectl version`): v1.13.5
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift): kubeadm
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/master/ceph-toolbox.html)): Health OK

### PR DESCRIPTION
**Pull request by rajatsingh25aug**
[skip ci] Changing the metadataPool and dataPools replicated size from 1 to 2 has no effect.

Signed-off-by: RAJAT SINGH <rajasing@redhat.com>

**Description of your changes:**
The old Filesystem is not getting updated when changed with the new File system.

**Which issue is resolved by this Pull Request:**
Resolves #3144 

**Checklist:**

- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [ ] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](https://github.com/rook/rook/blob/master/CONTRIBUTING.md#comments)
- [ ] Add the flag for skipping the CI if this PR does not require a build. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for more details.

### CODE REVIEW

File: pkg/operator/ceph/file/controller.go [hunk 1]
```diff
@@ -224,6 +225,10 @@ func (c *FilesystemController) filesystemOwner(fs *cephv1.CephFilesystem) metav1
 }
 
 func filesystemChanged(oldFS, newFS cephv1.FilesystemSpec) bool {
+	if cmp.Equal(oldFS.MetadataPool.Replicated.Size, newFS.MetadataPool.Replicated.Size) {
```
Comment (rajatsingh25aug) [inline pkg/operator/ceph/file/controller.go]: @travisn will this be a good practice to use `cmp.Equals`?.
Also, if I try to change the replicas in `filesystem.yaml` after making above changes in the go code (hoping changes will be reflected), the changes are not reflected at all.
Can you please drop some more hints maybe?.
Thanks

Comment (travisn) [reply pkg/operator/ceph/file/controller.go]: This basic comparison is preferring for ints:
```
oldFS.MetadataPool.Replicated.Size == newFS.MetadataPool.Replicated.Size
```

Comment (travisn) [reply pkg/operator/ceph/file/controller.go]: In the method `doFilesystemCreate()` looks like the filesystem will not be updated if it already exists. So we'll need to change this to update the pools if the size changed.

**Comment by rajatsingh25aug** [comment]
Screenshot 2019-11-27 at 8 01 11 PM
Ok So I made changes requested. What I did is that I replaced the entire `oldFs` with `newFs` data and that's why you see this message. Will update this aswell.

**Pull request closed without merging**